Nos/Controller/NoteEditorController.swift

```diff
@@ -0,0 +1,256 @@
+    private func insertMention(note: String, range: NSRange) {
+        insert(text: note.prefix(10).appending("..."), link: "nostr:\(note)", at: range)
+    }
+    
+    /// Inserts a string at the given range and adds the link attribute with the given `link`.
+    private func insert(text: String, link: String, at range: NSRange) {
+        guard let textView else {
+            return
+        }
+        
+        let linkAttributes = defaultNSAttributes.merging(
+            [NSAttributedString.Key.link: link], 
+            uniquingKeysWith: { _, key in key }
+        )
+        let link = NSAttributedString(
+            string: text,
+            attributes: linkAttributes
+        )
+        
+        let attributedString = textView.attributedText.mutableCopy() as! NSMutableAttributedString
+        attributedString.replaceCharacters(in: range, with: link)
+        textView.attributedText = attributedString
+        textView.selectedRange.location = range.location + link.length
+    }
+    
+    /// Takes the same arguments as `textView(_:shouldChangeTextIn:replacementText:)` and detects the case where the 
+    /// user is typing inside a link (like a mention of another author). If this is the case we remove the link 
+    /// attribute from the link and insert the given `newText` and return true. Otherwise this will return `false`.
+    private func removeLinkAttributesUnderCursor(
+        selectedRange: NSRange, 
+        in textView: UITextView, 
+        newText: String
+    ) -> Bool {
+        let attributedText = textView.attributedText!
+        if attributedText.length > 0, selectedRange.location < attributedText.length || newText.isEmpty {
+            var rangeOfLink = NSRange()
+            let location = newText.isEmpty ? max(selectedRange.location, 0) : selectedRange.location
+            let link = attributedText.attribute(
+                .link, 
+                at: location, 
+                longestEffectiveRange: &rangeOfLink, 
+                in: NSRange(location: 0, length: attributedText.length)
+            )
+            
+            if link != nil {
+                let newAttributesString = (attributedText.mutableCopy() as! NSMutableAttributedString)
+                newAttributesString.removeAttribute(.link, range: rangeOfLink)
+                newAttributesString.replaceCharacters(in: selectedRange, with: newText)
+                textView.attributedText = newAttributesString
+                textView.selectedRange = NSRange(
+                    location: selectedRange.location + (newText as NSString).length, 
+                    length: 0
+                )
+                return true
+            }
+        }
+        
+        return false
+    }
+        
+    /// Call this when the user has typed a '@' and it will trigger the mentions autocomplete to open if appropriate.
+    /// This function will return `true` if it opened the mentions autocomplete.
```

Comment:
I always like to see the `- Returns:` to explain what a func returns since it makes it nicer to read the doc comment in the popup. 

```suggestion
    /// - Returns: `true` if the mentions autocomplete was opened.
```

(check my syntax, but I think that's it)